webpack/assets/javascripts/react_app/components/common/forms/ForemanForm/ForemanForm.js

```diff
@@ -18,49 +26,78 @@ export const prepareErrors = errors =>
 export const isInitialValid = ({ validationSchema, initialValues }) =>
   !validationSchema ? true : validationSchema.isValidSync(initialValues);
 
-const ForemanForm = props => (
-  <Formik
-    onSubmit={(values, actions) =>
-      props.onSubmit(values, actions).catch(exception => {
-        actions.setSubmitting(false);
-        actions.setErrors(prepareErrors(exception.errors));
-      })
-    }
-    initialValues={props.initialValues}
-    validationSchema={props.validationSchema}
-    isInitialValid={isInitialValid}
-    enableReinitialize={props.enableReinitialize}
-  >
-    {formProps => {
-      const disabled = formProps.isSubmitting || !formProps.isValid;
+const ForemanForm = ({
+  onSubmit,
+  onSuccess,
+  children,
+  initialValues,
+  validationSchema,
+  enableReinitialize,
+  onCancel,
+  item,
```

Comment:
Should we fail fast somehow on missing `item` and let developers know it is required?